File: web/src/enums/nodeType.ts
export enum NodeType {
  Start = 'Start',
  Input = 'Input',
  LLM = 'LLM',
  Intent = 'Intent',
  Dataset = 'Dataset',
  Output = 'Output',
  Http = 'Http',
  WebSearch = 'WebSearch',
  Text = 'Text',
  Sql = 'Sql',
  Ask = 'Ask',
}

export function getNodeTypeName(nodeType: NodeType) {
  switch (nodeType) {
    case NodeType.Start:
      return '开始'
    case NodeType.Input:
      return '输入'
    case NodeType.LLM:
      return '大模型'
    case NodeType.Intent:
      return '意图识别'
    case NodeType.Dataset:
      return '知识库'
    case NodeType.Output:
      return '输出'
    case NodeType.Http:
      return 'HTTP 调用'
    case NodeType.WebSearch:
      return '联网搜索'
    case NodeType.Sql:
      return 'SQL 查询'
    case NodeType.Text:
      return '文本处理'
    case NodeType.Ask:
      return '提问'
    default:
      return '节点'
  }
}

File: ai/enums/node_type.py
from enum import Enum


class NodeType(str, Enum):
    """
    Enumeration of node types.
    """

    # Chatflow
    START = "Start"
    INPUT = "Input"
    OUTPUT = "Output"
    DATASET = "Dataset"
    LLM = "LLM"
    INTENT = "Intent"
    ASK = "Ask"
    WEBSEARCH = "WebSearch"
    TEXT = "Text"
    SQL = "Sql"
    HTTP = "Http"

    @classmethod
    def get_display_names(cls) -> dict[str, str]:
        return {
            cls.START: "开始",
            cls.INPUT: "输入",
            cls.OUTPUT: "输出",
            cls.DATASET: "知识库",
            cls.LLM: "大模型",
            cls.INTENT: "意图识别",
            cls.ASK: "提问",
            cls.WEBSEARCH: "联网搜索",
            cls.TEXT: "文本处理",
            cls.SQL: "SQL 查询",
            cls.HTTP: "HTTP 调用",
        }

    @property
    def display_name(self) -> str:
        """
        Returns the display name of the node type.
        """
        return self.get_display_names().get(self.value, self.value)
